utils/src/main/java/io/atomix/utils/config/ConfigMapper.java

```diff
@@ -558,20 +573,24 @@ protected static String toSetterName(String name) {
   protected static class SetterDescriptor {
     private final String name;
     private final Method setter;
+    private final boolean deprecated;
 
     SetterDescriptor(String name, Method setter) {
       this.name = name;
       this.setter = setter;
+      this.deprecated = setter.getAnnotation(Deprecated.class) != null;
```

Comment:
nice!

Reply:
I know, I really like that feature. We need to log a lot better information about configuration options. Another nice feature would be making suggestions for invalid properties, e.g. using Levenshtein distance to match a real property or something. `Did you mean 'cluster.members.1.id'?`